client/my-sites/checkout/src/components/checkout-main-content.tsx

```diff
@@ -274,35 +278,36 @@ function CheckoutSidebarNudge( {
 	}
 
 	/**
-	 * TODO !hasMonthlyProduct can likely be removed after checkout v2 is merged
-	 * V2 checkout handles monthly products in the CheckoutSidebarPlanUpsell so this condition is not needed
+	 * TODO !hasMonthlyProduct can likely be removed after Jetpack refactors their sidebar nudge
+	 * to account for monthly products like CheckoutSidebarPlanUpsell does
 	 */
-	if ( ! hasMonthlyProduct || shouldUseCheckoutV2 ) {
-		return (
-			<CheckoutSidebarNudgeWrapper>
-				<CheckoutSidebarPlanUpsell />
-				<JetpackAkismetCheckoutSidebarPlanUpsell />
-				{ ( isPurchaseRenewal || domainWithoutPlanInCartOrSite ) && (
-					<SecondaryCartPromotions
-						responseCart={ responseCart }
-						addItemToCart={ addItemToCart }
-						isCartPendingUpdate={ isCartPendingUpdate }
-						isPurchaseRenewal={ isPurchaseRenewal }
-					/>
-				) }
-				{ shouldUseCheckoutV2 && (
-					<CheckoutSummaryFeaturedList
-						responseCart={ responseCart }
-						siteId={ siteId }
-						isCartUpdating={ FormStatus.VALIDATING === formStatus }
-						onChangeSelection={ changeSelection }
-					/>
-				) }
-			</CheckoutSidebarNudgeWrapper>
-		);
-	}
 
-	return null;
```

Comment:
Not sure if removing this final null is a good idea, from what I can tell most use cases are handled here 🤔